src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -217,13 +217,22 @@ object ProtocPlugin extends AutoPlugin with Compat {
   private[this] def unpack(
       deps: Seq[File],
       extractTarget: File,
-      log: Logger
+      streams: TaskStreams
   ): Seq[(File, Seq[File])] = {
-    IO.createDirectory(extractTarget)
+    def cachedExtractDep(dep: File): Seq[File] = {
+      val cached = FileFunction.cached(streams.cacheDirectory / dep.name) { deps =>
```

Comment:
given the signature of the closure wrapped by FileFunction.cached and because we need to keep the mapping of extracted protos per dependency, the cache has to be per dependency